test/entt/entity/observer.cpp

```diff
@@ -369,3 +369,34 @@ TEST(Observer, GroupCornerCase) {
     ASSERT_FALSE(add_observer.empty());
     ASSERT_TRUE(remove_observer.empty());
 }
+
+enum class CustomEntity : std::uint32_t
+{
+};
+
+
+// This could be used to add custom methods to the registry or, in case of non-public inheritance,
+// to restrict the access to the registry.
+class CustomRegistry: public entt::basic_registry<CustomEntity> {
+};
+
+template<typename Component>
```

Comment:
Not a fan of puting new types in the middle of a test file because I need them there.
I usually have all types at the top of the test file, then I use them as needed.